src/input/input_json.rs

```diff
@@ -84,18 +85,29 @@ impl<'a> Input<'a> for JsonValue {
         }
     }
 
-    fn strict_str(&'a self) -> ValResult<EitherString<'a>> {
+    fn exact_str(&'a self) -> ValResult<EitherString<'a>> {
         match self {
+            // Justification for `strict` instead of `exact` is that in JSON strings can also
+            // represent other datatypes such as UUID and date more exactly, so string is a
+            // converting input
             JsonValue::Str(s) => Ok(s.as_str().into()),
             _ => Err(ValError::new(ErrorTypeDefaults::StringType, self)),
         }
     }
-    fn lax_str(&'a self, coerce_numbers_to_str: bool) -> ValResult<EitherString<'a>> {
+
+    fn validate_str(
+        &'a self,
+        strict: bool,
+        coerce_numbers_to_str: bool,
+    ) -> ValResult<ValidationMatch<EitherString<'a>>> {
+        // Justification for `strict` instead of `exact` is that in JSON strings can also
```

Comment:
will this want to change in future, if so can we leave a consistent `TODO: V3` comment to make it easy to find in future.

Reply:
This depends what you think of e.g. https://github.com/pydantic/pydantic/issues/7097. The current semantic that this PR preserves is that `UUID` is "as good" as `str` for a UUID-format JSON string, so it's dependent on ordering. Changing this to be "exact" means that in JSON mode a `str` in a union wins over `UUID`, `datetime`, etc, irrespective of position.

That is a positive change for simplicity and matches Python semantics better, so there is a case to consider breaking this, but it's not clear cut IMO.

Reply:
I think it's hard to argue that in `datetime | str` or `UUID | str` or `datetime | int` that the type which actually matches JSON type shouldn't take priority. If you agree I would be happy to change it now.

Otherwise add the `TODO: V3` comment so we remember to change it in future.

Reply:
I agree but feeling a bit cautious so will mark as TODO and we can decide later if it's a 2.6 or V3 thing.